features/storage/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -209,6 +213,140 @@ void test_multi_threads()
     TEST_ASSERT_EQUAL(0, err);
 }
 
+void test_get_erase_value()
+{
+    utest_printf("\nTest BlockDevice::get_erase_value()..\n");
+
+    BlockDevice *block_device = BlockDevice::get_default_instance();
+    TEST_ASSERT_NOT_NULL(block_device);
+
+    int err = block_device->init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Check erase value
+    int erase_value = block_device->get_erase_value();
+    if (erase_value < 0) {
+        utest_printf("\nblock_device->get_erase_value() returned %d, which means the erase value is unknown\n", erase_value);
+        goto exit;
+    }
+
+    // Assuming that get_erase_value() returns byte value
+    //  "If get_erase_value returns a non-negative byte value"
+    TEST_ASSERT(erase_value < 255);
+
+    // Allocate buffer for read test data
+    bd_addr_t start_address = (rand() * block_device->get_erase_size()) % (block_device->size());
+    bd_size_t read_buf_size = block_device->get_erase_size();
+    uint8_t *read_buf = (uint8_t*)malloc(read_buf_size);
+    TEST_ASSERT_NOT_NULL(read_buf);
+
+    // Erase given memory region
+    utest_printf("\nerasing erased memory region");
+    err = block_device->erase(start_address, read_buf_size);
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Read erased memory region
+    utest_printf("\nreading erased memory region");
+    err = block_device->read((void*)read_buf, start_address, read_buf_size);
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Verify erased memory region
+    utest_printf("\nverifying erased memory region");
+    for (bd_size_t i=0; i<read_buf_size; i++) {
+        TEST_ASSERT_EQUAL(erase_value, read_buf[i]);
+    }
+
+    free(read_buf);
+
+exit:
+    // BlockDevice deinitialization
+    err = block_device->deinit();
+    TEST_ASSERT_EQUAL(0, err);
+}
+
+void test_contiguous_erase_write_read()
+{
+    // BlockDevice initialization
+    utest_printf("\nTest Contiguous Erase/Program/Read Starts..\n");
+
+    BlockDevice *block_device = BlockDevice::get_default_instance();
+    TEST_ASSERT_NOT_NULL(block_device);
+
+    // Initialize BlockDevice
+    int err = block_device->init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Target write size that is not too large but requires more than 1 erase block.
+    bd_size_t data_buf_size = 5000;
+    data_buf_size -= data_buf_size % block_device->get_erase_size(); // Align size to BlockDevice erase size
+    utest_printf("\ndata_buf_size=%" PRIu64 "", (uint64_t)data_buf_size);
+    TEST_ASSERT(data_buf_size > 1000);
+
+    // Allocate write buffer
+    uint8_t *data_buf = (uint8_t*)malloc(data_buf_size);
+    TEST_ASSERT_NOT_NULL(data_buf); // Check that allocation succeeded
+
+    // Check that test contains more than 1 block
+    int block_count = data_buf_size / block_device->get_erase_size();
+    utest_printf("\nblock_device->get_erase_size()=%d", block_device->get_erase_size());
+    utest_printf("\nblock_count=%d", block_count);
+    TEST_ASSERT(block_count >= 2); // Check that more than 1 block is used
```

Comment:
I'm not entirely sure I follow the test logic, but on BD such as SPIF or QSPIF typically erase_size is 4096 bytes or larger. For that case, using data_buf_size = 5000, I think you will end up with (block_count = 1)

Reply:
Good input. Should I increase the buffer size to 10k? My worry here is that on some restricted device, we don't have enough RAM available.

Reply:
Maybe you can allocate "2 x get_erase_size()" or however you need for the test. And skip the test for restricted devices using something like: 
https://github.com/ARMmbed/mbed-os/blob/0dbdd9ee0bd8e16683140e6de698bbffa186428a/features/storage/TESTS/blockdevice/general_block_device/main.cpp#L127